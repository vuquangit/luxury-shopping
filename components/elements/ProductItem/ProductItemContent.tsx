import { FC } from 'react'

type Props = {
  title: string
  subtitle?: string
  content_link?: string
}

const ProductItemContent: FC<Props> = ({ title, subtitle, content_link }) => {
  return (
    <div className="home-product__content">
      <div className="home-product__content--title">{title}</div>
      {subtitle && (
        <div className="home-product__content--subtitle">{subtitle}</div>
      )}
      {content_link && (
        <div className="home-product__content--link">
          <div className="btn btn--primary content-link-btn">
            {content_link}
          </div>
        </div>
      )}
    </div>
  )
}

export default ProductItemContent
