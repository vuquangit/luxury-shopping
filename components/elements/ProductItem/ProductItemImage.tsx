import { FC } from 'react'
import LazyImage from 'components/elements/LazyImage'

type Props = {
  image: string
  aspectRatio?: [number, number]
  isBackgroundImage?: boolean
}

const ProductItemImage: FC<Props> = ({
  image,
  aspectRatio = [1, 1],
  isBackgroundImage = false,
}) => (
  <>
    {image && (
      <LazyImage
        src={image}
        aspectRatio={aspectRatio}
        isBackgroundImage={isBackgroundImage}
        className={'home-product__image'}
      />
    )}
  </>
)

export default ProductItemImage
