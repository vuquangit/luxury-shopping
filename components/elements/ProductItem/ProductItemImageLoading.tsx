import { FC } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faSpinner } from '@fortawesome/free-solid-svg-icons'

// type Props = {
//   aspectRatio?: [number, number]
// }

const ProductItemImageLoading: FC = () => {
  // eslint-disable-next-line no-console
  console.log('loading....')

  const aspectRatio = [10, 10]

  const paddingTop = `${(aspectRatio[1] / aspectRatio[0]) * 100}%`

  return (
    <div className="image-loading" style={{ paddingTop }}>
      <div className="image-loading__content">
        <FontAwesomeIcon
          icon={faSpinner}
          className="image-loading__content--icon"
        />
      </div>
    </div>
  )
}

export default ProductItemImageLoading
