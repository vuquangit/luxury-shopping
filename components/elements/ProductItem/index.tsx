import { FC } from 'react'
import Loadable from 'react-loadable'
// import ProductItemImage from './ProductItemImage'
import ProductItemContent from './ProductItemContent'
import Loading from './ProductItemImageLoading'

type Props = {
  id: number
  image: string
  title: string
  link: string
  subtitle?: string
  content_link?: string
  aspectRatio?: [number, number]
}

const ProductItem: FC<Props> = ({
  image,
  link,
  title,
  subtitle,
  content_link,
  aspectRatio = [1, 1],
}) => {
  const LoadableComponent = Loadable({
    loader: () => import('./ProductItemImage'),
    loading: Loading,
  })

  return (
    <a href={link} className="home-product">
      <LoadableComponent
        image={image}
        aspectRatio={aspectRatio}
        isBackgroundImage
      />
      <ProductItemContent
        title={title}
        subtitle={subtitle}
        content_link={content_link}
      />
    </a>
  )
}

export default ProductItem
