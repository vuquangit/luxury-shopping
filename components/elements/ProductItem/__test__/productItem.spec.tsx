import React from 'react'
import renderer from 'react-test-renderer'
import ProductItem from '../index'
import productData from './mockData.json'

it('renders product item', () => {
  const tree = renderer.create(<ProductItem {...productData} />).toJSON()
  expect(tree).toMatchSnapshot()
})
