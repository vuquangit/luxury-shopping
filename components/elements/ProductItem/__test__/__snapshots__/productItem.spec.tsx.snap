// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders product item 1`] = `
<a
  className="home-product"
  href="/women/handbags"
>
  <div
    className="lazyImage"
    style={
      Object {
        "paddingTop": "100%",
      }
    }
  >
    <div
      className="lazyImage__img home-product__image"
      style={
        Object {
          "backgroundImage": "url(https://res.cloudinary.com/dnzsa2z7b/image/upload/v1601808438/luxury-shoping/home-page/new-products/93842498_1376660662534221_6427143203815686144_o_lm1fzo.jpg)",
        }
      }
    />
  </div>
  <div
    className="home-product__content"
  >
    <div
      className="home-product__content--title"
    >
      PETITE MALLE SOUPLE EMPREINTE
    </div>
    <div
      className="home-product__content--subtitle"
    >
      Shop the Exclusive Prelaunch Now
    </div>
    
  </div>
</a>
`;
